src/index.js

```diff
@@ -1,26 +1,287 @@
-//add any imports if needed, or write your script directly in this file.
-//import SomePackage from "PackageName";
-
-//make sure to export main, with the signature
-export function main (el, service, imEntity, state, config) {
-    // sample code here to convert the provided intermine object (e.g. perhaps
-    // an id) into an identifier the tool expects. e.g.:
-    // of course if your tool was built for intermine it might understand
-    // intermine ids already, but many others tools expect a gene symbol or
-    // protein accession, etc...
-
-/***************************************************************************
-/****** Example - you can delete this and replace with your own code *******
-
-    //protVista expects an accession, so convert intermine id to accession
-
-    var columnToConvert = config.columnMapping[imEntity.class][imEntity.format];
-    var accession = new imjs.Service(service)
-        .findById(imEntity.class, imEntity.value)
-        .then(function(response) {
-        //put some code here to initialise your tool.
+/* Overview of how we find the homologues of a gene:
+ * 1. We only have the internal (to this mine) ID of our gene, so we run a
+ *    query to get the symbol and organism for our gene.
+ * 2. Using the API URL of our mine passed from BlueGenes, we get its namespace
+ *    and use this to identify the mine in a query for mine instances to find
+ *    its neighbours.
+ * 3. For each neighbour, (or rather neighbourhood) we will query each of the
+ *    belonging mine instances for homologues, gradually building up the list
+ *    of results.
+ */
+
+// Constants you can change to customise the behaviour of this tool.
+var SHOW_GENES_COUNT = 5;
+var QUERY_TIMEOUT = 30*1000;
+
+// State we keep track of so we don't need to query the DOM.
+var nsToElem = {};
+var emptyMines = [];
+var noteElem;
+
+// Use the gene ID we get from BlueGenes to get the gene's symbol and organism.
+function querySymbol(url, id) {
+  return new Promise(function(resolve) {
+    var intermine = new imjs.Service({root: url});
+
+    var query = {
+      "from": "Gene",
+      "select": [
+        "symbol",
+        "primaryIdentifier",
+        "secondaryIdentifier",
+        "organism.name"
+      ],
+      "orderBy": [
+        {
+          "path": "symbol",
+          "direction": "ASC"
+        }
+      ],
+      "where": [
+        {
+          "path": "id",
+          "op": "=",
+          "value": id,
+          "code": "A"
+        }
+      ]
+    };
+
+    intermine.records(query).then(function(res) {
+      const symbol = geneToSymbol(res[0]);
+      const organism = res[0].organism.name;
+      resolve([symbol, organism]);
     });
+  });
+}
+
+// Use our mine's API URL to get its neighbours.
+function queryNeighbours(mineUrl) {
+  return new Promise(function(resolve) {
+    fetch(
+      "http://registry.intermine.org/service/namespace?url="
```

Comment:
let's extract the registry base url to the top of the file to make it easily editable and only used once, and also make it https